src/mlpack/core/metrics/non_maximal_supression.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file non_maximal_supression.hpp
+ * @author Kartik Dutt
+ *
+ * Definition of Non Maximal Supression metric.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_METRICS_NMS_HPP
+#define MLPACK_CORE_METRICS_NMS_HPP
+
+namespace mlpack {
+namespace metric {
+
+/**
+ * Definition of Non Maximal Supression.
+ *
+ * Performs non-maximal suppression (NMS) on the boxes according to their
+ * intersection-over-union (IoU). NMS iteratively removes lower scoring boxes
+ * which have an IoU greater than threshold with another high scoring box.
+ *
+ * For bounding box representation there are two common representation
+ * either as coordinates i.e. each value in vector represents a
+ * coordinate in the format x0, y0, x1, y1 where x0, y0 represent the
+ * lower left coordinate and x1, y1, represent upper right coordinate.
+ * 
+ * Second representation follows the following representation : x0, y0, h, w.
+ * Where x0 and y0 are bottom left bounding box coordinates and h, w are
+ * height and width of the bounding box.
+ *
+ * @tparam useCoordinates Toggles between the two representation of bounding box.
```

Comment:
Let's rectify the `tparam`'s name and add descriptions about `Transpose`. BTW, do you think users will use the `Transpose` parameter? otherwise, would it be a little unnecessary option?
```suggestion
 * @tparam UseCoordinates Toggles between the two representation of bounding box.
```

Reply:
Sure, I'll make that change as well. About the `Transpose` parameter, I am not sure, if they use with rest of the layers the current implementation would work and transpose is not needed. I am not 100% if users would need it, I can remove it if you want. Kindly let me know what you think.
Regards.

Reply:
I think this idea is great, but it would be better to remove it at this point. We will add this option later when we realized it's vital on the API side. Sometimes, it's hard to remove the existing feature than add new feature in consideration of backward compatibility after release.

Reply:
Sure, Then I'll remove it and fix the style issue and add more tests. Then I guess this will be ready. Thanks a lot.